src/ser/serde.rs

```diff
@@ -248,13 +251,28 @@ impl ser::Serializer for Serializer {
     #[inline]
     fn serialize_newtype_struct<T: ?Sized>(
         self,
-        _name: &'static str,
+        name: &'static str,
         value: &T,
     ) -> crate::ser::Result<Bson>
     where
         T: Serialize,
     {
-        value.serialize(self)
```

Comment:
serialization works by serializing a newtype with a name that this serializer can recognize (`UUID_NEWTYPE_NAME`). When it sees this type, it uses the output of `uuid::Uuid`'s `Serialize` implementation (for the human readable serializer,  a string) and turns it into a subtype 4 Binary.

The same thing is done for the raw serializer, except that it expects `uuid::Uuid` to produce bytes.